controller/common/manage_children.go

```diff
@@ -45,10 +45,45 @@ func ApplyUpdate(orig, update *unstructured.Unstructured) (*unstructured.Unstruc
 	if err != nil {
 		return nil, err
 	}
+	// Revert metadata fields that are known to be read-only, system fields.
+	// We should never try to update those, even if the user asks us to.
+	// See: https://github.com/GoogleCloudPlatform/metacontroller/issues/76
+	if err := revertObjectMetaSystemFields(newObj, orig); err != nil {
+		return nil, fmt.Errorf("failed to revert ObjectMeta system fields: %v", err)
+	}
 	dynamicapply.SetLastApplied(newObj, update.UnstructuredContent())
 	return newObj, nil
 }
 
+// objectMetaSystemFields is a list of JSON field names within ObjectMeta that
```

Comment:
Should we put this to a global file?

Reply:
What do you mean by a global file? I put it here for now because I like to keep things close to where they're being used if they're only used once. We can always move it later.

I looked around in Kubernetes itself to see if there was any existing list or code to revert ObjectMeta system fields, but I didn't find anything reusable. It looks like they just hard-coded the steps to set the fields in various places.

Reply:
Oh, i mean a file that records all hard-coded attributes/structs. +1 on keeping things close if they are only used once.